tests/MongoDB.Driver.Core.Tests/Core/Authentication/MongoAWSAuthenticatorTests.cs

```diff
@@ -0,0 +1,188 @@
+            var expectedClientFirstMessage = new BsonDocument()
+                .Add("r", new BsonBinaryData(clientNonce))
+                .Add("p", new BsonInt32('n'));
+            var expectedClientSecondMessage = new BsonDocument()
+                .Add("a", authHeader)
+                .Add("d", timestamp);
+
+            var serverFirstMessage = new BsonDocument()
+                .Add("s", new BsonBinaryData(serverNonce))
+                .Add("h", new BsonString(host));
+            var saslStartReply = MessageHelper.BuildReply<RawBsonDocument>(RawBsonDocumentHelper.FromJson(
+                $"{{conversationId: 1, done: false, payload: BinData(0,\"{ToBase64(ToBytes(serverFirstMessage))}\"), ok: 1}}"));
+            var saslContinueReply = MessageHelper.BuildReply<RawBsonDocument>(RawBsonDocumentHelper.FromJson(
+                "{conversationId: 1, done: true, payload: BinData(0,\"\"), ok: 1}"));
+
+            var subject = new MongoAWSAuthenticator(credential, null, randomByteGeneratorMock.Object, dateTimeProviderMock.Object);
+
+            var connection = new MockConnection(serverId);
+            connection.EnqueueReplyMessage(saslStartReply);
+            connection.EnqueueReplyMessage(saslContinueReply);
+
+            var expectedRequestId = RequestMessage.CurrentGlobalRequestId + 1;
+
+            Exception exception;
+            if (async)
+            {
+                exception = Record.Exception(() => subject.AuthenticateAsync(connection, connectionDescription, CancellationToken.None).GetAwaiter().GetResult());
+            }
+            else
+            {
+                exception = Record.Exception(() => subject.Authenticate(connection, connectionDescription, CancellationToken.None));
+            }
+            exception.Should().BeNull();
+
+            SpinWait.SpinUntil(() => connection.GetSentMessages().Count >= 2, TimeSpan.FromSeconds(5)).Should().BeTrue();
+
+            var sentMessages = MessageHelper.TranslateMessagesToBsonDocuments(connection.GetSentMessages());
+            sentMessages.Count.Should().Be(2);
+
+            var actualRequestId0 = sentMessages[0]["requestId"].AsInt32;
+            var actualRequestId1 = sentMessages[1]["requestId"].AsInt32;
+            actualRequestId0.Should().BeInRange(expectedRequestId, expectedRequestId + 10);
+            actualRequestId1.Should().BeInRange(actualRequestId0 + 1, actualRequestId0 + 11);
+
+            var expectedFirstMessage = GetExpectedSaslStartMessage(actualRequestId0, expectedClientFirstMessage);
+            var expectedSecondMessage = GetExpectedSaslContinueMessage(actualRequestId1, expectedClientSecondMessage);
+
+            sentMessages[0].Should().Be(expectedFirstMessage);
+            sentMessages[1].Should().Be(expectedSecondMessage);
+        }
+
+        private static string GetExpectedSaslContinueMessage(int requestId, BsonDocument clientMessage)
+        {
+            return  "{" +
+                        "opcode: \"query\", " +
```

Comment:
Just an FYI: An alternative style to multiline bsondocuments, but feel free to simply resolve this after reading this, I just wanted to share another option :) :
https://github.com/MikalaiMazurenka/mongo-csharp-driver/blob/8925eaeb11dbb35702c2a417bdb1d9c25903c8f3/tests/MongoDB.Driver.Core.Tests/Core/Authentication/ScramSha256AuthenticatorTests.cs#L251